### PR TITLE
[WIP] Add MarianMT to models exportable with ONNX

### LINKED ISSUE #13823: Add MarianMT to models exportable with ONNX
# 🚀 Feature request

Add the support to convert the MarianMT model with the `transformers.onnx` package documented [here](https://huggingface.co/transformers/serialization.html). The conversion now returns `marian () is not supported yet. Only [...] are supported. If you want to support (marian) please propose a PR or open up an issue.`

## Motivation

MarianMT is one of the best translation models in the hub because of the extensive number of pretrained language pairs, but it can be slow for real-time use cases. A conversion to ONNX combined with quantization could significantly improve inference time.

### PR DESCRIPTION
**Pull request by Maxinho96**
Resolves #13823

@patil-suraj @michaelbenayoun

### CODE REVIEW

**Review by michaelbenayoun — COMMENTED** [review]
Thanks for your PR!

File: src/transformers/models/marian/modeling_marian.py [hunk 1]
```diff
@@ -1132,6 +1132,13 @@ def forward(
 
             >>> last_hidden_states = outputs.last_hidden_state
         """
+        # different to other models, Marian automatically creates decoder_input_ids from
```
[the remark this answers is not in the document]
Comment (patil-suraj) [reply src/transformers/models/marian/modeling_marian.py:1135]: This shouldn't be added here. This is done for BART because of the denoising pre-training objective, so only BART and BART prepare the `deocder_input_ids` from `input_ids`.  For Marian users should pass either `decoder_input_ids` or `labels` and this is already handled by the `MarianMTModel` class.

[the remark this answers is not in the document]
Comment (Maxinho96) [reply src/transformers/models/marian/modeling_marian.py:1135]: > This shouldn't be added here. This is done for BART because of the denoising pre-training objective, so only BART and mBART prepare the `deocder_input_ids` from `input_ids`. For Marian users should pass either `decoder_input_ids` or `labels` and this is already handled by the `MarianMTModel` class.

Thank you @patil-suraj for the reply, but as I said in [this](https://github.com/huggingface/transformers/issues/13823#issuecomment-932349647) comment, `transformers.onnx.convert.export()` calls `MarianMTModel.forward()` without passing `decoder_input_ids` or `labels`, so how is this supposed to be handled?
